src/main/java/hudson/plugins/gradle/injection/MavenExtensionDownloadHandler.java

```diff
@@ -0,0 +1,97 @@
+package hudson.plugins.gradle.injection;
+
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.common.StandardUsernamePasswordCredentials;
+import hudson.Util;
+import hudson.plugins.gradle.injection.extension.ExtensionClient;
+import jenkins.model.Jenkins;
+
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class MavenExtensionDownloadHandler implements MavenInjectionAware {
```

Comment:
I tested this logic manually, but I'm going to add a test as part of this PR for this specific class

Reply:
Test added